### PR TITLE
Resolve #490 Major Refactor

### LINKED ISSUE #490: Major Refactor
Currently, there is no unified interface for passing and handling transfer data, which causes problems whilewriting unit tests.

### PR DESCRIPTION
**Pull request by timothywangdev**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -16,6 +16,6 @@ REACT_APP_ETHEREUM_NETWORK=rinkeby
 REACT_APP_INFURA_API_KEY=9775236247814824bc231e65b1d4972a
 REACT_APP_BTC_NETWORK=testnet
 REACT_APP_DAI_ADDRESS=0xDb29d7f3973e1A428f0578705e7eA1632F2E4aC5
-REACT_APP_CHAINSFER_API_ENDPOINT=https://3zowl9pdcl.execute-api.us-east-1.amazonaws.com/Prod
+REACT_APP_CHAINSFER_API_ENDPOINT=https://ovj5je8dt4.execute-api.us-east-1.amazonaws.com/Prod
```
[the remark this answers is not in the document]
Comment (timothywangdev) [reply .env]: fixed

File: src/containers/ReceiveWalletSelectionContainer.jsx [hunk 2]
```diff
@@ -67,30 +67,30 @@ class ReceiveWalletSelectionContainer extends Component {
...
     }
   }
 
-  onSync = (cryptoType: string) => {
-    const { syncLedgerAccountInfo } = this.props
-    syncLedgerAccountInfo(cryptoType, 0, (index, change) => { this.setState({ syncProgress: { index, change } }) })
+  onSync = () => {
+    let { wallet, walletSelection, transfer } = this.props
+    let accounts = wallet.crypto[transfer.cryptoType]
+    console.log(this.props)
```
[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/containers/ReceiveWalletSelectionContainer.jsx]: fixed

File: src/wallets/bitcoin.js [hunk 3]
```diff
@@ -160,7 +160,7 @@ export default class WalletBitcoin implements IWallet<WalletDataBitcoin, Account
     }
   }
 
-  sync = async (progress: any) => {
+  sync = async (progress: Function) => {
```
[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/wallets/bitcoin.js]: fixed

File: src/components/RecipientComponent.jsx [hunk 4]
```diff
@@ -89,7 +89,7 @@ class RecipientComponent extends Component<Props> {
     const decimals = getCryptoDecimals(cryptoSelection)
     if (name === 'transferAmount') {
       if (!validator.isFloat(value, { min: 0.001, max: utils.toHumanReadableUnit(balance, decimals) })) {
-        if (value === '-' || parseFloat(value) < 0.001) {
+        if (value === '-' || parseFloat(value) < 0.00001) {
```
[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/components/RecipientComponent.jsx]: fixed

File: src/actions/walletActions.js [hunk 5]
```diff
@@ -1,117 +1,60 @@
...
-          }
-        }
-      }
-    }
-
+    let wallet = WalletFactory.createWallet(WalletUtils.toWalletData('metamask', cryptoType, []))
+    await wallet.retrieveAddress()
     // listen for accounts changes
     window.ethereum.on('accountsChanged', function (accounts) {
       dispatch(onMetamaskAccountsChanged(accounts))
```
Comment (InVincible2016) [inline src/actions/walletActions.js]:
```js
    window.ethereum.on('accountsChanged', (accounts) => {
      let newWallet = WalletFactory.createWallet(WalletUtils.toWalletData('metamask', cryptoType, accounts))
      dispatch(onMetamaskAccountsChanged(newWallet.getWalletData()))
    })
```

**Pull request merged**